### PR TITLE
docs: add example uses of tokens

### PR DESCRIPTION
**Pull request by shadowspawn**
Add example files using tokens to add behaviour that has come up in discussions.

The token support is added in #129. That PR currently includes these examples for illustration, but I have added a separate PR so the examples may be discussed separately from landing the code.

See: #84
Closes: #52

### CODE REVIEW

File: examples/no-repeated-options.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+'use strict';
+
+// How might I throw if an option is repeated?
```
Comment (Eomm) [inline examples/no-repeated-options.js:3]:
```suggestion
// Q: How might I throw if an option is repeated?
```

File: examples/limit-long-syntax.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+'use strict';
+
+// How might I require long options with values use '='?
```
Comment (Eomm) [inline examples/limit-long-syntax.js:3]:
```suggestion
// Q: How might I require long options with values use '='?
```

File: examples/ordered-options.mjs [hunk 3]
```diff
@@ -0,0 +1,35 @@
+// Now might I enforce that two flags are specified in a specific order?
```
Comment (Eomm) [inline examples/ordered-options.mjs:1]:
```suggestion
// Q: Now might I enforce that two flags are specified in a specific order?
```

File: examples/no-repeated-options.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+// 1. const { parseArgs } = require('node:util'); // from node
+// 2. const { parseArgs } = require('@pkgjs/parseargs'); // from package
+const { parseArgs } = require('..'); // in repo
+
+const options = {
+  ding: { type: 'boolean', short: 'd' },
+  beep: { type: 'boolean', short: 'b' }
```
Comment (Eomm) [inline examples/no-repeated-options.js:11]: By using `multiple: true` and checking that the array value has only 1 item is this an option?

Or do we aim to use `tokens` only?

File: examples/no-repeated-options.js (hunk 1 above)
Comment (shadowspawn) [reply examples/no-repeated-options.js:3]: Thanks for comments. I don't think adding a Q is appropriate here, as not Q&A style. But it does make me wonder if using a brief question is a weak way of introducing the example. I was thinking of it as some context, or a problem, or a challenge. And perhaps too concise/lazy!

I'll try reworking into a more explicit setting for the example.

File: examples/no-repeated-options.js (hunk 4 above)
Comment (shadowspawn) [reply examples/no-repeated-options.js:11]: Yes could do using `multiple`. 
But also yes,  this was intended to be an example of using tokens. (And I could say that in an expanded introduction...)

**Comment by shadowspawn** [comment]
Updated the comments to clarify the examples which are mainly about doing something interesting with tokens, and not about solving a problem as such.

**Review by Eomm — APPROVED** [review]

**Pull request merged**